src/main/java/ua/utility/kfsdbupgrade/ForeignKeyReference.java

```diff
@@ -105,7 +105,13 @@ private String buildIndexName(int indx) {
 		StringBuilder retval = new StringBuilder(128);
 		retval.append(schemaName.toUpperCase());
 		retval.append(".");
-		retval.append(indexNameTemplate.replace("[table-name]", tableName).replace("{index}", "" + indx));
+		// Oracle before v 12.2 only allows identifiers upto 30 characters in length, so shorten index names
+		String shortTableName = tableName;
```

Comment:
Good point. ORM was not even on my radar regarding this. Agreed that Mods are the High priority. DBUpgrade Low priority. I guess I should shelve, research an revisit this one. Thanks.